### ISSUE TITLE
[kit-routes] Support array search params

### ISSUE THREAD

**Issue by fehnomenal** [jycouet/kitql#580]
**Is your feature request related to a problem? Please describe.**

I want to configure the following:
```
          explicit_search_params: {
            calculationIds: { type: 'number[]', required: true },
            workshopId: { type: 'string', required: true },
          },
```

but this fails in the generated file because `appendSp` converts all params to strings:

![image](https://github.com/jycouet/kitql/assets/9959940/ebaa9c4a-6a83-469b-afad-7bb95e183f09)

![image](https://github.com/jycouet/kitql/assets/9959940/c82da59a-a753-4b88-bad5-d18b09716d8b)

**Describe the solution you'd like**

`appendSp` should also accept `string[]` and `number[]` and append them. I guess something like this should work:
```ts
  const mapping = Object.entries(sp).flatMap(([name, val]) => {
    if (val == undefined) return [];
    if (Array.isArray(val)) return val.map((v) => [name, String(v)]);
    return [[name, String(val)]];
  });
```

**Describe alternatives you've considered**

Not setting the search params explicitely.

**Additional context**

Thank you very much for this plugin :hugs:

**Comment by jycouet** [comment]
Sounds legit 😉
Will do it soon

**Comment by jycouet** [comment]
Acutally, the only thing is to add types to the function, like https://github.com/jycouet/kitql/pull/583/files#diff-428c41f044580b0012cca0f191f227b016994088c35ee306a1ad47d70419ae23R186 No?

It's already adding a coma between values. (https://github.com/jycouet/kitql/pull/583/files#diff-33237037534b3bac096fcb3f8659e1b83232daf3002470f75d37f3f9d7104c07) So the `mapping` function is not needed.

I think that it's all good with this PR https://github.com/jycouet/kitql/pull/583 but I want to make sure I understand all of it.

Please let me know before I merge and publish 👍

**Comment by jycouet** [comment]
@all-contributors please add @fehnomenal for example

**Comment by fehnomenal** [comment]
> Acutally, the only thing is to add types to the function, like [#583 (files)](https://github.com/jycouet/kitql/pull/583/files#diff-428c41f044580b0012cca0f191f227b016994088c35ee306a1ad47d70419ae23R186) No?
> 
> It's already adding a coma between values. ([#583 (files)](https://github.com/jycouet/kitql/pull/583/files#diff-33237037534b3bac096fcb3f8659e1b83232daf3002470f75d37f3f9d7104c07)) So the `mapping` function is not needed.
> 
> I think that it's all good with this PR #583 but I want to make sure I understand all of it.
> 
> Please let me know before I merge and publish 👍

Yeah you're right, it does not crash, I got misled by the type errors.

But there is a difference between `?ids=1&ids=2&ids=3` and `?ids=1%2C2%2C3`. If I'd use this to generate internal links I could adapt the handling to the comma separated list. But I use this for external `LINKS` over which I have no control :sweat_smile: 

I opened #585 as an alternative approach. If you find it to complex just close it :smile:

**Comment by jycouet** [comment]
OMG, I didn't know it was allowed to have multiple `ids` in SearchParams! 🤔
Is it common?

I like the idea of your PR by supporting multiple use-cases ✅

**Comment by jycouet** [comment]
Should be available in `vite-plugin-kit-routes@0.3.0`
Thx for your great work @fehnomenal 🎉
